lib/spack/spack/test/util/util_url.py

```diff
@@ -36,49 +37,58 @@ def test_url_parse():
 
     parsed = url_util.parse('file://path/to/resource')
     assert(parsed.scheme == 'file')
-    assert(parsed.netloc == '')
-    expected = os.path.abspath(os.path.join('path', 'to', 'resource'))
+    expected = convert_to_posix_path(
+        os.path.abspath(
+            posixpath.join('path', 'to', 'resource')))
+    if is_windows:
+        expected = expected.lstrip('C:')
     assert(parsed.path == expected)
 
+    if is_windows:
+        parsed = url_util.parse('file://C:\\path\\to\\resource')
+        assert(parsed.scheme == 'file')
+        expected = '/' + convert_to_posix_path(
+            posixpath.join('path', 'to', 'resource')
```

Comment:
Out of curiosity, can `posixpath.join` produce a path that would need to be formatted with `convert_to_posix_path`? What does it change about the path?

I see a similar call a few lines above although in that case it makes sense why it might be needed because there's an "intervening" call to `os.path.abspath`.